packages/contracts/test/contracts/utils/ContractAddressGenerator.spec.ts

```diff
@@ -1,48 +1,43 @@
 import '../../setup'
 
 /* External Imports */
-import { getLogger } from '@eth-optimism/core-utils'
-import { createMockProvider, deployContract, getWallets } from 'ethereum-waffle'
-import { utils, Contract } from 'ethers'
+import { ethers } from '@nomiclabs/buidler'
+import { utils, Contract, ContractFactory, Signer } from 'ethers'
 
 /* Internal Imports */
 import { create2Tests } from '../../test-helpers/data/create2.test.json'
-import { DEFAULT_ETHNODE_GAS_LIMIT } from '../../test-helpers/core-helpers'
 import { buildCreate2Address } from '../../test-helpers'
 
-/* Contract Imports */
-import { ContractAddressGeneratorContractDefinition } from '../../../src'
-
-/* Logging */
-const log = getLogger('contract-address-generator', true)
-
 /* Tests */
 describe('ContractAddressGenerator', () => {
-  const [wallet1, wallet2] = getWallets(
-    createMockProvider({ gasLimit: DEFAULT_ETHNODE_GAS_LIMIT })
-  )
-  let contractAddressGenerator: Contract
+  let wallet1: Signer
+  let wallet2: Signer
+
+  before(async () => {
+    ;[wallet1, wallet2] = await ethers.getSigners()
+  })
 
+  let ContractAddressGenerator: ContractFactory
   beforeEach(async () => {
-    contractAddressGenerator = await deployContract(
-      wallet1,
-      ContractAddressGeneratorContractDefinition,
-      [],
-      {
-        gasLimit: DEFAULT_ETHNODE_GAS_LIMIT,
-      }
+    ContractAddressGenerator = await ethers.getContractFactory(
+      'ContractAddressGenerator'
     )
   })
 
+  let contractAddressGenerator: Contract
+  beforeEach(async () => {
+    contractAddressGenerator = await ContractAddressGenerator.deploy()
+  })
+
   describe('getAddressFromCREATE', async () => {
     it('returns expected address, nonce: 1', async () => {
       const nonce = 1
       const expectedAddress = utils.getContractAddress({
-        from: wallet1.address,
+        from: await wallet1.getAddress(),
```

Comment:
woah interesting that the get address is async! is that a buidler thing?

Reply:
It's a little weird. Buidler uses `Signer` accounts by default. I ended up having to sort of reimplement the `getWallets()` behavior anyway, so I may switch everything back to use `Wallet` instances later on.